### PR TITLE
Implemented riscv64 syscall support by modifying source code of musl

### PR DESCRIPTION
**Pull request by Sssssaltyfish**
More precisely speaking, by replacing all usage of `ecall` in musl with a direct function to `riscv_syscall_asm`, which dispatches the corresponding syscalls to their handling procedures.

Besides this modification, this PR does the following jobs:

1. modified building procedure of musl to separate different builds using different `ARCH` and `MODE` (might could be cherry-picked), and to auto-patch the musl source code (note that it won't be auto-restored when `ARCH` is different, so a redownload of musl source is always needed).

2. changed the default size of physical memory to 1G, as more would cause dtb page error in my environment.

### CODE REVIEW

**Comment by coolyjg** [comment]
I created another branch named `riscv64-musl`. Please put this implementation`(modify musl)` into that branch. Thanks!

File: platforms/riscv64-qemu-virt.toml
Comment (coolyjg) [inline platforms/riscv64-qemu-virt.toml:1]: Same reason as another PR. If this is changed to 1G, then X64 and A64 should do the same work.

File: scripts/make/build_musl.mk [hunk 1]
```diff
@@ -3,11 +3,15 @@ rust_lib := target/$(TARGET)/$(MODE)/lib$(rust_lib_name).a
...
-build_dir := $(muslibc_dir)/build_musl_$(ARCH)
+build_dir := $(muslibc_dir)/build_musl$(build_suffix)
+install_dir_name := install$(build_suffix)
+install_dir := $(muslibc_dir)/$(install_dir_name)
+
 musl_dir := $(muslibc_dir)/musl-$(musl_version)
-inc_dir := $(muslibc_dir)/install/include
-c_lib := $(muslibc_dir)/install/lib/libc.a
+inc_dir := $(install_dir)/include
+c_lib := $(install_dir)/lib/libc.a
```
Comment (coolyjg) [inline scripts/make/build_musl.mk:14]: Please apply these changes to #69 too! The `install_dir` should be distinguished by `ARCH` too. Thanks!

**Comment by coolyjg** [comment]
Please rebase into one commit! Thanks!

**Comment by Sssssaltyfish** [comment]
I guess it is ready to be merged. The extra commits exist because I merged upstream/dev, so I guess they'll disappear if upstream/riscv64-musl is also up-to-date with it.

**Pull request merged**